airflow/models/dagrun.py

```diff
@@ -124,7 +124,7 @@ class DagRun(Base, LoggingMixin):
         ),
     )
 
-    task_instances = relationship(TI, back_populates="dag_run")
+    task_instances = relationship(TI, back_populates="dag_run", cascade='all, delete, delete-orphan')
```

Comment:
Not sure the best option for us:
@uranusjr @ashb @kaxil 
https://docs.sqlalchemy.org/en/14/orm/cascades.html#unitofwork-cascades